### PR TITLE
Bug fix/internal issue #650

### PR DESCRIPTION
**Pull request by Dronplane**
Added analyzer cache update on coordinator befor view create/update.
If cluster is just started analyzers cache is empty on coordinator. And attempt to create/update ArangoSearch view will fail due to not found analyzers. Now cache is reread for current database and system database (as analyzers from system database could be used also in other databases).

Fix is verified by js tests

https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/10054/

### CODE REVIEW

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 1]
```diff
@@ -1907,6 +1907,25 @@ Result IResearchAnalyzerFeature::cleanupAnalyzersCollection(irs::string_ref cons
   return {};
 }
 
+Result IResearchAnalyzerFeature::loadAvailableAnalyzers(irs::string_ref const& dbName) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:1910]: maybe rename to `reloadAnalyzers(...)`?

File: arangod/RestHandler/RestViewHandler.cpp [hunk 2]
```diff
@@ -204,8 +205,18 @@ void RestViewHandler::createView() {
   }
 
   try {
+    // First refresh our analyzers cache to see all latest changes in analyzers
+    auto res = server().getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
+                         .loadAvailableAnalyzers(_vocbase.name());
```
Comment (gnusi) [inline arangod/RestHandler/RestViewHandler.cpp:210]:
```suggestion
                       .loadAvailableAnalyzers(_vocbase.name());
```

File: arangod/RestHandler/RestViewHandler.cpp [hunk 3]
```diff
@@ -274,6 +285,14 @@ void RestViewHandler::modifyView(bool partialUpdate) {
       return;
     }
 
+    // First refresh our analyzers cache to see all latest changes in analyzers
+    auto const analyzersRes = server().getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
+      .loadAvailableAnalyzers(_vocbase.name());
```
Comment (gnusi) [inline arangod/RestHandler/RestViewHandler.cpp:290]:
```suggestion
    auto const analyzersRes = server().getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
                                                            .loadAvailableAnalyzers(_vocbase.name());
```

File: arangod/V8Server/v8-views.cpp [hunk 4]
```diff
@@ -186,8 +187,18 @@ static void JS_CreateViewVocbase(v8::FunctionCallbackInfo<v8::Value> const& args
                                                            header.slice(), false, true);
 
   try {
+
+    // First refresh our analyzers cache to see all latest changes in analyzers
+    TRI_GET_GLOBALS();
+    auto res = v8g->_server.getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
+      .loadAvailableAnalyzers(vocbase.name());
```
Comment (gnusi) [inline arangod/V8Server/v8-views.cpp:194]:
```suggestion
    auto res = v8g->_server.getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
                                          .loadAvailableAnalyzers(vocbase.name());
```

File: arangod/V8Server/v8-views.cpp [hunk 5]
```diff
@@ -579,7 +590,16 @@ static void JS_PropertiesViewVocbase(v8::FunctionCallbackInfo<v8::Value> const&
       TRI_V8_THROW_EXCEPTION(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND);
     }
 
-    auto res = view->properties(builder.slice(), partialUpdate);
+    auto& vocbase = GetContextVocBase(isolate);
+    TRI_GET_GLOBALS();
+    auto res = v8g->_server.getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
+      .loadAvailableAnalyzers(vocbase.name());
```
Comment (gnusi) [inline arangod/V8Server/v8-views.cpp:596]:
```suggestion
	    auto res = v8g->_server.getFeature<arangodb::iresearch::IResearchAnalyzerFeature>()
                                            .loadAvailableAnalyzers(vocbase.name());
```

**Comment by Dronplane** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10055/

**Pull request merged**